src/test/java/org/folio/patronblocks/repository/UserSummaryRepositoryImplTest.java

```diff
@@ -0,0 +1,154 @@
+import io.vertx.core.Future;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class UserSummaryRepositoryImplTest extends APITests {
+
+  private UserSummaryRepository userSummaryRepository = new UserSummaryRepositoryImpl(
+    PostgresClient.getInstance(vertx, "test_tenant"));
+  private static final String USER_SUMMARY_TABLE = "user_summary";
+
+  @Before
+  public void setUp(TestContext context) {
+    Async async = context.async();
+    PostgresClient.getInstance(vertx, OKAPI_TENANT)
+      .delete(USER_SUMMARY_TABLE, new Criterion(), event -> {
+        if (event.failed()) {
+          log.error(event.cause());
+          context.fail(event.cause());
+        } else {
+          async.complete();
+        }
+      });
+  }
+
+  @Test
+  public void shouldAddUserSummary() {
+    String userId = UUID.randomUUID().toString();
+    Future<String> userSumId = userSummaryRepository.saveUserSummary(
+      createUserSummary(userId, BigDecimal.ONE, 2, 1));
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSumId.result() != null);
+
+    assertThat(userSumId.result(), is(userId));
+
+    Future<Optional<UserSummary>> userSummaryById =
+      userSummaryRepository.getUserSummaryById(userId);
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSummaryById.result().isPresent());
+
+    UserSummary resultUserSummary = userSummaryById.result().get();
+
+    assertThat(resultUserSummary.getId(), is(userId));
+    assertThat(resultUserSummary.getNumberOfLostItems(), is(2));
+    assertThat(resultUserSummary.getNumberOfOpenFeesFinesForLostItems(), is(1));
+    assertThat(resultUserSummary.getOutstandingFeeFineBalance(), is(BigDecimal.ONE));
+  }
+
+  @Test
+  public void shouldUpdateUserSummary() {
+    String userId = UUID.randomUUID().toString();
+    Future<String> userSumId = userSummaryRepository.saveUserSummary(
+      createUserSummary(userId, new BigDecimal(2), 4, 2));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSumId.result() != null);
+
+    Future<Boolean> isUpdatedFuture = userSummaryRepository.updateUserSummary(
+      createUserSummary(userId, new BigDecimal(10), 3, 1));
+
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(isUpdatedFuture::result);
+
+    Future<Optional<UserSummary>> userSummaryById =
+      userSummaryRepository.getUserSummaryById(userId);
+    Awaitility.await()
+      .atMost(1, TimeUnit.SECONDS)
+      .until(() -> userSummaryById.result().isPresent());
+
+    UserSummary updatedUserSummaryRecord = userSummaryById.result().get();
+    assertThat(updatedUserSummaryRecord.getNumberOfLostItems(), is(3));
+    assertThat(updatedUserSummaryRecord.getNumberOfOpenFeesFinesForLostItems(), is(1));
+    assertThat(updatedUserSummaryRecord.getOutstandingFeeFineBalance(), is(BigDecimal.TEN));
```

Comment:
It was created as new BigDecimal(10) and now we're comparing it to BigDecimal.TEN. Can we make it more unified?